### ISSUE TITLE
some general feedback from a newbie

### ISSUE THREAD

**Issue by Schnobs** [KSP-RO/RP-0#146]
First off, to get you an idea where I'm coming from: you may know me as "Laie" on the forum. I consider myself to be quite knowledgeable and skilled when it comes to basic/stockish KSP. I came to RO out of curiosity; NathanKell suggested that I should try RP-0 as it might introduce me to RO in a more gradual manner.

I tried to follow the getting started guide, https://github.com/KSP-RO/RP-0/wiki/Getting-Started-Tutorial

The first thing that bit me was KCT. I have only the most cursory knowledge of what it is and tries to achieve, so I was more than happy to be given detailed instructions about what to do first, namely "Sink all your points into the first VAB rate." I don't know what I did wrong or maybe I shouldn't have spent *all* points; at any rate, my first tiny sounding rocket then had a construction time of ~150 days. Time warp is slightly different than what I'm used to, so I overshot this by another 50d. When I went to the pad, the rocket wasn't ready (WTF?) -- I only had the option to do a simulation. At this point, I restarted my career with KCT off.

There's very detailed instructions about how to build your first rocket, yet I managed to shoot myself in the foot regardless. I am quite familiar with procedural wings (not B9) and still had a hard time to make them small enough as to not look completely out of place. So I opted for the stock AV-T1 winglet that I scaled down. Launching the rocket at a 5 degree angle didn't end well. It took me some time to figure out that these winglets add substantial weight. It's really pwings or nothing.

That construction guide introduced me to procedural fairings. I wondered why you go to such great lengths when any old nosecone would do, but then realized that figuring out how to set up fairings would have been another obstacle later on. Thanks for doing that. 

Quite generally, the sheer number of mods that goes into RO is a problem in and of itself, IMO. RemoteTech, KCT, TAC -- each of them would be quite a mouthful on their own, and I'm in a situation where I have to learn all of them at the same time.

I specifically wonder wether "Part Test" is such a good idea. I had a bad spell of running on reduced thrust, I think. I'm not entirely sure. I went back to the VAB to triple check my vessel, launched again, failed again, restarted KSP... only later did it occur to me that this may have been plain old honest equipment failure. Later on, the boosters on my first atlas didn't ignite and I started looking for fuel flow issues. When equipment fails on it's own, it should at least deliver a pop-up message.

I have similar stories about RemoteTech.

Back to topic: Tilting the rocket by 5 degrees still isn't the best of ideas. It introduces the concept of gravity turn, which is good, but crossing the 24km mark is quite difficult. I went through several rounds of altitude records by pointing it straight up, though.

Pretty soon, I developed a routine and set ever higher altitude records. The WAC Corporal worked a treat, probably because the engine was thoroughly tested and reliable at that point. Still, launches quickly started to require ten minutes or more, with absolutely nothing to do but sit and watch. Sounding rockets were a good opportunity to develop a sense for what a good launch looks like, how tilt and TWR affect your flight path, but that turned old pretty quickly. On top of that, the learning experience was disturbed by lots of part failures. If I ever start over, I hope I can skip these early baby steps, or go over them in a few big leaps (though Part test will probably prevent this).

As you can probably guess, I'm not too fond of that mod. And I seriously wonder what kind of person enjoys playing with Part Test *and* KCT.

Personally, I'm more in favor of going on with the game. I'd strongly recommend to start the campaign with maximum funds, which allows one to upgrade the more important facilities to level-2 from the get-go (Action Groups! More than two contracts! Rockets of decent size and weight!).

**Comment by aw94** [comment]
As for your last paragraph -- did you try the modified .dll mentioned in the RP-0 getting started guide?

(I may have some more to say once I digest what you said)

**Comment by Schnobs** [comment]
Ooops, nope. I didn't try that dll.

Other than that, "digest" sounds scary. Don't take my words too seriously. I tried to be very verbose about how and where I struggled, and a best guess as to why, but what it boils down to is that I didn't read all the docs before I started.

I went into RP-0 in the hopes that it might be kind of a tutorial, which may be wrong at the outset.

That's probably also the main reason why Flight Test has been putting me off as much as it did -- learning is much easier if every fault is my own.

**Comment by jwvanderbeck** [comment]
As the author of TestFlight I would honestly say, **don't** use it until
you are comfortable with the game.  It is designed to add more realism, and
realism by its nature makes things harder.

On the flip side, I definitely do need to improve communication to the
player when things go wrong.  Most all failures should show up in the
TestFlight Master Status Display, so if you are unsure of if a part failed
or you did, look there.  Personally I always have it open during flight.
Some failures, like ignition fails and exploded parts, might not show up
there but should **always** show up in the standard KSP Flight Log (F3).

KCT, RT, and TF are all more advanced realism mods so maybe not the best
way to get introduced to things.  That said maybe we can make that better
with improved documentation.

- John Vanderbeck
- http://www.johnvanderbeck.com

**Comment by aw94** [comment]
Sorry, didn't intend to be scary; I was sort of jumping from thing to thing at the time, and didn't get to give it the read it deserved.

You do raise good points. RP-0/RO is a lot to swallow, especially with all the suggested mods installed too. Not everyone enjoys some of the more "hard-core" mods are, as you found out.

It'd be interesting if we could have a mode or something that starts with more stock-ish behavior (within limits), and gradually adds in more realism until it's at full-out RO/RP-0, but I don't know if that's possible. I'll ask the channel sometime, but with 1.0 on the horizon, it might be a bit before that gets any real consideration. In the meantime, just remove the mods you don't want. The forum post should have something indicating whether the mod makes things harder -- those are probably the ones you'll want to look at first.

As for early game, yeah, RP-0 is a bit weak on that side of things. That's being worked on, though, so hopefully some time in the future there'll be enough variety to keep things interesting, which should make the early game not so painful.

Not sure about TF pop-up messages on failures; the closest thing I can think of is the small HUD accessed through the TF toolbar button that you can keep on screen; however, some other messages (e.g. failure to ignite) only show up in the F3 log.

Might consider adding some pictures to the wiki too, if that would help make things clearer. And I'm not sure about the wing mass thing; theoretically, if a pwing looks like a regular wing, it should cost/mass about the same, but it seems that's not always the case. If you TweakScale'd it, though, perhaps it's that, since that seems to mess with a few things.

**Comment by NathanKell** [comment]
@Schnobs  I've been busy with various stuff today so don't have a proper response yet other than *thanks so much*! We really appreciate feedback. :)

**Comment by OtherBarry** [comment]
I literally just woke up (Yay Weekends) so I too will have to come back to
this.

Just wanted to add that the issue with tweakscale and wings (or anything
else) is that once you hit launch, the tweakscaled item's size stays the
same, but reverts back to it's original mass. So for sounding rockets this
can lead to your wings being heavier than the rest of the rocket. Also
tweakscaled thing's don't work too well in FAR.

On the difficulty note, I think RP-0 is on different ends in different
areas. It's great for teaching you how to launch with FAR and deal with
realistically scaled orbital mechanics, but were it not for my hundreds of
hours playing with sandbox RO, the new mods and ship design would have
completely stumped me.

As for TestFlight and KCT, I'd say that for beginners, and most people,
they don't add much fun to start with. RP-0 is pretty hard without having
to work out how to use KCT, and especially TestFlight. It usually takes me
at least 5 simulations to sort out all the kinks in my rocket, having
random failures would make that even harder. I'd suggest a strong warning
on the forum post, and maybe putting them in the recommended section on
ckan if they're not already.

Also, @NathanKell, ping me next weekend and I'll try and write up some more
tutorials.

**Comment by NathanKell** [comment]
Regarding KCT, it's possible the interface changed. In the KCT I'm used to, there are two build rates, the first VAB rate and the second VAB rate. A full upgrade should give you 0.85bp/sec on the prime VAB rate, which means sounding rockets should only take a day or three (assuming something like 400? funds cost in the editor). However, once a rocket is ready, as the tutorial says you have to roll it out to the pad, and that takes a bit longer (this is because you can have many rockets in storage "ready to go" but only one on the pad).

Regarding five degrees--as you mention, five degrees is a bit much, but I expected would still be little enough to allow decent heights. And it's much easier to talk about hitting shift-A then try to walk someone through turning snap off and then tilting only a degree or two (imprecisely) with the rotation gizmo...but if five degrees is too much, the latter is what it will be.

Regarding fins: as OtherBarry mentions, tweakscale has some issues with fins. We'll ask Crzymdm to lower the minimums on the B9 wings, I agree they're hard to get small enough for sounding rockets.

The reason for Proc Fairings was both that (hehe), but also so that the instruments can be aerodynamically shielded. If you use a nosecone, there's nowhere to place the instruments except on the outer surface of the rocket.

I forgot about how brutal TestFlight can be early--that needs a much more prominent place in the tutorial, indeed the tutorial should have first gone through the 'ground test' stage. Or the R&D bit if @jwvanderbeck has added that. It might also be worth considering having the WAC sustainer count as pre-tested--it comes unlocked, so one would expect it to be ready for use.

@aw94 pictures would be awesome! I wrote the tutorial purely from memory in about 20 minutes in between 1.0-related busyness, so it's rather sparse, but I wanted to have something to point @Schnobs towards as s/he was just trying out RP-0 (and indeed RO) for the first time.

**Comment by NathanKell** [comment]
Indeed, the more I think about it the more I like having parts that start unlocked start fully-Data'd. @jwvanderbeck is that possible to do?

**Comment by OtherBarry** [comment]
If it is, might be something to link with KCT research time. Eg, if you
have KCT research time enabled, then by default the parts that come with
that are fully TestFlight tested/data'd.

**Comment by aw94** [comment]
@NathanKell I'll see if I can get pictures tomorrow. Also, don't suppose an in-game "transition" from stock to RO would be possible without a major rewrite of KSP, would it? That would be a great way to let people try RO/RP-0 if it can be done

**Comment by NathanKell** [comment]
@OtherBarry I meant the parts in the start node, which come all ready to be used. When you research a node, normally you then have to purchase the parts within it; the start node, all the parts in it are pre-purchased. Those are the only parts I'm talking about. That said, if and when @jwvanderbeck implements non-flight data increases, that might be worth considering applying to parts in currently-being-researched nodes...?

@aw94: Not really, no, KSP can't really handle dynamically loading and unloading all those mods.

**Comment by aw94** [comment]
:/

**Comment by OtherBarry** [comment]
Ohh right. Start parts definitely.

**Comment by jwvanderbeck** [comment]
There is currently no method of starting a part with data, though it wouldn't be hard to add.  I've been out of it the past month after starting the new job (Just have no energy to DO anything lol) but i'll see if I can ease myself back into some KSP work this weekend and add that.  I need to do the big 1.3 release too.

**Comment by NathanKell** [comment]
@jwvanderbeck  that'd be awesome, we miss you lots! (And we want to use RealScience for RP-0 1.0 :) )

That said, much rather have you employed than KSPing. :)

**Comment by jwvanderbeck** [comment]
You know, what you could actually do is just patch the parts so that it is
max reliability at 0 data.

- John Vanderbeck
- http://www.johnvanderbeck.com

**Comment by Schnobs** [comment]
@everyone: thanks! I was afraid I might wrong-foot you.

Regarding the 5 degrees, a picture with the rotate gizmo open, and the words "tilt *slightly*" would probably be better than however many words. Then again, just leaving it to the player to figure out may not be the worst. She will have to solve more complicated riddles soon enough.

--

Flight Test Test and KCT already are in the recommended (or is it suggested?) section. I definitely had to check a box to get them.

No way I would go through *all* the mods' README files and other documentation before I started playing, but it's been a few days now and I've done some catching up. Give it another week and I couldn't even tell you why it required so much determination to get into RP-0.

In hindsight, it seems that I should have looked into Flight Test before anything else. That one completely blindsided me (I'm sooo used to any failure being my fault) and it's relatively inconspicuous. At least when compared to KCT, which I immediately recognized as not suitable for my purpose and uninstalled right away.

--

I wonder about the many parts that are not supported by RO, RP-0, or both. If they are unsupported because noone came around to rewriting the config, I hereby offer to help. If they're just not supported and never will be, I'd suggest to add recipes to the part pruner and advertise it more aggressively.

**Comment by Schnobs** [comment]
I've tried the *Assembly-CSharp.dll* mentioned above, that's supposed to update the VAB/Launchpad limits. Indeed it does, but as a side effect, several parts became unclickable in the VAB.

I could place the thermometer alright, but not (re)move it later, nor select it when setting up action groups. Same with barometer and the aerobee sustainer engine.

**Comment by NathanKell** [comment]
@Schnobs yeah, rotate gizmo + pic sounds like the correct solution there. As for TF and KCT I think it is worth continuing to recommend them, but to rather improve the RP-0 tutorial (and do the engine reliability change described above--so the tutorial can cover testing once you unlock your first new engine).

Regarding assembly-CSharp, I have literally never heard of that issue before from anyone using it. The only similar issue I've heard of has come from Texture Replacer. Do you have TR installed? If so, try uninstalling it and see if the problem goes away. What OS are you using? If it's not Windows I can try making the same changes to the OS-specific DLL, in case they're different? (it's Mono, they shouldn't be, and if they are that isn't the error you'd hit, but who knows.)

**SERVICES ACCEPTED** -- we'd dearly love the help! They are indeed just not touched yet, and flagged as non-RP0 precisely to goad folks (hi @pjf ) into patching them. :)

**Comment by jwvanderbeck** [comment]
Not being able to click on things in the VAB implies something is throwing
NREs, and that would be best tracked down with a log

- John Vanderbeck
- http://www.johnvanderbeck.com

**Comment by pjf** [comment]
### CKAN metadata

Looking at our CKAN metadata, we really should be moving some mods from `recommends` to `suggests`. 'Recommends' means the mod *can* operate without those packages, but it's unusual to see that for a typical user, whereas 'suggests' means a user may or may not have those packages typically installed.

RemoteTech, Engine Ignitor, Test Flight, KCT, DMagic Orbital Science, VSR, etc should all be "suggests". This will give new users an easier time, *and* is consistent with our mission statement of RP-0 being a *lightweight* career mode for RealismOverhaul. I'd also downgrade pFairings to a recommends; you *can* play without it (using fairings from AIES, etc), but it would be weird. (c.f. ModuleManager and RealismOverhaul, which RP-0 is utterly broken without).

**Comment by NathanKell** [comment]
@pjf then I think there needs to be another category. :P
Because you will *not* be playing "RP-0 as intended" and "RP-0 as it's been balanced for" and "RP-0 as the tutorials are written for" if you don't have those mods installed. You will be able to play it--it won't break, in the way it would if they were depends--but they are a good bit up from "meh, maybe".
DMagic certainly can go to suggests (although some distinction should be made between it, which is supported, and NovaPunch, which is suggested by RO but not supported in RP-0), but Ven's Stock Revamp, well, RO *does* expect you to use it; the engines will look very weird otherwise. And same for the others, the balance is designed with them in mind. Now, as I'm sure you're already thinking, MM can get around that to some degree...but the tutorials should be written for the "full experience" IMO.

**Comment by NathanKell** [comment]
@Schnobs: @aw1621107 confirmed that appears to be caused by Texture Replacer interacting poorly with Ven's Stock Revamp.
